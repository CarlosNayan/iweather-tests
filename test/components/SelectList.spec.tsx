import { SelectList } from "@components/SelectList";
import { render, fireEvent, screen } from "@testing-library/react-native";

describe("Component: SelectList", () => {
  let data = [
    { id: "1", name: "City 1", longitude: 0, latitude: 0 },
    { id: "2", name: "City 2", longitude: 0, latitude: 0 },
  ];

  it("should render correctly", () => {
    const { getByText } = render(
      <SelectList data={data} onChange={jest.fn()} onPress={jest.fn()} />
    );

    expect(getByText(/city 1/i)).toBeTruthy();
  });

  it("should be able to select a city", () => {
    const onPress = jest.fn();

    render(<SelectList data={data} onChange={jest.fn()} onPress={onPress} />);

    fireEvent.press(screen.getByText(/city 1/i));

    expect(onPress).toHaveBeenCalledTimes(1);
    expect(onPress).toHaveBeenCalledWith(data[0]);
  });

  it("not should be show options when data props is empty", () => {
    render(<SelectList data={[]} onChange={jest.fn()} onPress={jest.fn()} />);

    expect(screen.queryByTestId("options")).toBeNull();
    expect(screen.queryByTestId("options-container").children).toHaveLength(0);
  });
});
