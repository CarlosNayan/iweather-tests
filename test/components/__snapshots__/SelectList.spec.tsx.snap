// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SelectList should render correctly 1`] = `
<View
  style={
    {
      "position": "relative",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#22222F",
        "borderRadius": 8,
        "flexDirection": "row",
        "fontFamily": "Nunito_400Regular",
        "fontSize": 16,
        "height": 56,
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
  >
    <TextInput
      onChangeText={[MockFunction]}
      placeholderTextColor="#7F7F98"
      style={
        {
          "color": "#FFFFFF",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      {
        "borderRadius": 8,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 64,
      }
    }
    testID="options-container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="options"
    >
      <Text
        style={
          {
            "backgroundColor": "#3B3B54",
            "borderBottomColor": "#22222F",
            "borderBottomWidth": 1,
            "color": "#FAFAFA",
            "fontFamily": "Nunito_400Regular",
            "fontSize": 16,
            "paddingHorizontal": 20,
            "paddingVertical": 16,
          }
        }
      >
        City 1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="options"
    >
      <Text
        style={
          {
            "backgroundColor": "#3B3B54",
            "borderBottomColor": "#22222F",
            "borderBottomWidth": 1,
            "color": "#FAFAFA",
            "fontFamily": "Nunito_400Regular",
            "fontSize": 16,
            "paddingHorizontal": 20,
            "paddingVertical": 16,
          }
        }
      >
        City 2
      </Text>
    </View>
  </View>
</View>
`;
